crates/codegen/runtime/generator/src/lib.rs

```diff
@@ -25,13 +26,21 @@ pub enum OutputLanguage {
 struct ModelWrapper {
     rendering_in_stubs: bool,
     model: Option<RuntimeModel>,
+    builtin_labels: Vec<String>,
```

Comment:
I suggest adding `builtin_labels` to the `RuntimeModel` struct instead. This wrapper is only used for rendering the templates, and shouldn't contain language-related fields.

Reply:
I tried that at first, but for the stubs generation there is no model (see below). We may just skip outputting the built-in labels for the stubs? Not sure they are used for anything.

Reply:
Oh, yes, they are used in `parser_support/precedence_helper.rs`

Reply:
I moved it into `RuntimeModel` which made the use of `Option<>` for the model unnecessary. Makes me think this may also allow removing _some_ of the template conditionals on `rendering_in_stubs`, but I'm not sure if it's better to leave them as is for clarity.

Reply:
Could we separate this drive-by refactoring of extracting Labels to a common crate into a dedicated PR? I'm refactoring parser codegen and touch a lot of related code; I'd love to rebase/upstream this smaller PR rather than potentially block here.

I think we can do the discussed side refactoring in a separate PR, if that's okay, to focus on the original issue 🙏

Reply:
Happy to extract the refactor to a separate PR, unless this can be merged as it is now. But you're right, this thread has already diverted too much from the original issue and it makes sense to discuss it separately.

Reply:
I don't mind either way. If it is going to impact the refactoring, we can also just push/merge the commit before that refactoring (https://github.com/NomicFoundation/slang/pull/976/commits/e6a72c07347c22efa84589af60a4e5c308c98e09?) for now if @ggiraldez is ok with it?

Reply:
Yes, no problem! I'll spin off a new PR with the last two commits.

Reply:
I opened #992 and left it as draft to be considered after this one and #991 are merged.